app/uk/gov/hmrc/customs/declaration/connectors/NrsConnector.scala

```diff
@@ -34,18 +33,16 @@ class NrsConnector @Inject()(http: HttpClient,
                              declarationConfigService: DeclarationsConfigService)
                             (implicit ec: ExecutionContext) {
 
-  private val XApiKey = "X-API-Key"
-
-  def send[A](nrsPayload: NrsPayload, apiVersion: ApiVersion)(implicit vpr: ValidatedPayloadRequest[A]): Future[NrSubmissionId] = {
+  def send[A](nrsPayload: NrsPayload, apiVersion: ApiVersion)(implicit vpr: ValidatedPayloadRequest[A], hc: HeaderCarrier): Future[NrSubmissionId] = {
     post(nrsPayload, declarationConfigService.nrsConfig.nrsUrl)
   }
 
-  private def post[A](payload: NrsPayload, url: String)(implicit vupr: ValidatedPayloadRequest[A]) = {
-
-    implicit val hc: HeaderCarrier = HeaderCarrier()
+  private def post[A](payload: NrsPayload, url: String)(implicit vupr: ValidatedPayloadRequest[A], hc: HeaderCarrier) = {
 
+    val nrsHeaders = Seq[(String, String)](("Content-Type", "application/json"), ("X-API-Key", declarationConfigService.nrsConfig.nrsApiKey))
+      .++ (hc.headers(List("Accept", "Gov-Test-Scenario")))
```

Comment:
So here we add 2 headers to what existed before. I understand why we add "Gov-Test-Scenario" but why do we add Accept too?

If they are both needed and both to facilitate the stubs, then we should make this clear in the code. Perhaps something like initialising a Seq called stubHeaders or something and then adding it.

Short of a complete refactor, we need to take great care to make it explicit what we are doing in this fragile piece of code. 

Actually, we are reusing (hc.headers(List("Accept", "Gov-Test-Scenario"))) at least twice, so it should be extracted out. This will reduce duplication and allow us to explain what we are doing with this.

Reply:
For stub to know which version to hit, it needs Accept header. Without that Accept, it will return 500 from the stub.
`Version: NO_VERSION_IN_HEADER
Message: No matching paths could be found
       ] 
2024-06-18 15:22:22,324 level=[INFO] logger=[uk.gov.hmrc.play.bootstrap.filters.DefaultLoggingFilter] thread=[application-pekko.actor.default-dispatcher-11] rid=[] user=[] message=[POST /declarations/submitdeclaration 500 11ms]`